community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/OffHeapByteArray.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.cache;
+
+import org.neo4j.unsafe.impl.internal.dragons.UnsafeUtil;
+
+public class OffHeapByteArray extends OffHeapNumberArray<ByteArray> implements ByteArray
+{
+    private final byte[] defaultValue;
+
+    protected OffHeapByteArray( long length, byte[] defaultValue, long base )
+    {
+        super( length, defaultValue.length, base );
+        this.defaultValue = defaultValue;
+        clear();
```

Comment:
i find it strange that we need to allocate memory for full array in the constructor and then feel it with default value, all of the cells :(